lib/git_blame.rb

```diff
@@ -26,7 +26,7 @@ def changes_since_last_green(build)
     def changes_in_branch(build)
       output = GitRepo.inside_repo(build.repository) do
         # TODO: Push this down into GitRepo and integration test it.
-        Cocaine::CommandLine.new("git log --cc --format='::!::%H|%cn <%ce>|%cd|%B::!::' 'origin/master..origin/#{build.branch_record.name}'").run
+        Cocaine::CommandLine.new("git log --cc --format='::!::%H|%cn <%ce>|%cd|%B::!::' 'master..#{build.branch_record.name}'").run
```

Comment:
After your comment I attempted to do something along these lines but I've given up. The kicker was that this line uses two dots in its git range and the git log line right below uses _three_ dots. That made it harder to abstract. Same with many other git log lines in this file.

I do believe this class could be improved quite a bit. Our homegrown method of parsing using "::!::" as a separator is bizarre.

However, the extra time spent looking at this file was fruitful. I found a nasty mistake in the command line flags for two of the methods: 071b9bc13a184346e58a39c1d521ebd81a84ef72